include/seastar/core/internal/io_sink.hh

```diff
@@ -58,19 +58,17 @@ public:
     // draining should try to drain more
     requires std::is_invocable_r<bool, Fn, internal::io_request&, io_completion*>::value
     size_t drain(Fn&& consume) {
-        size_t pending = _pending_io.size();
         size_t drained = 0;
+        auto it = _pending_io.begin();
 
-        while (pending > drained) {
-            pending_io_request& req = _pending_io[drained];
-
+        for (auto end = _pending_io.end(); it != end; ++it, ++drained) {
```

Comment:
This took some time to parse this loop :) Why not make it classical

```
    for (auto it = _pending_io.begin(); it != _pending_io.end(); it++) {
        ...
```

?

Reply:
Well, inside baseball: there was an intermediate version where I used `boost::deque` and where then after the loop `erase(begin(), it)` was called so `it` needed to survive the loop. Evidently, that's no longer the case as we use the count to do the erase, so I'm fixing that. In fact, this can just be a ranged-for.